src/Microsoft.Maintainability.Analyzers/Core/DoNotIgnoreMethodResults.cs

```diff
@@ -1,68 +1,146 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Semantics;
 
 namespace Microsoft.Maintainability.Analyzers
-{                   
+{
     /// <summary>
     /// CA1806: Do not ignore method results
     /// </summary>
-    public abstract class DoNotIgnoreMethodResultsAnalyzer : DiagnosticAnalyzer
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotIgnoreMethodResultsAnalyzer : DiagnosticAnalyzer
     {
         internal const string RuleId = "CA1806";
 
+        private static readonly ImmutableHashSet<string> s_stringMethodNames = ImmutableHashSet.CreateRange(
+            new string[] {
+                "ToUpper",
+                "ToLower",
+                "Trim",
+                "TrimEnd",
+                "TrimStart",
+                "ToUpperInvariant",
+                "ToLowerInvariant",
+                "Clone",
+                "Format",
+                "Concat",
+                "Copy",
+                "Insert",
+                "Join",
+                "Normalize",
+                "Remove",
+                "Replace",
+                "Split",
+                "PadLeft",
+                "PadRight",
+                "Substring",
+            });
+
```

Comment:
Do we have an issue tracking extending this to anything with [PureAttribute] applied?